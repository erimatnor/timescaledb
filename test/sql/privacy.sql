\c single :ROLE_SUPERUSER
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_privacy() RETURNS VOID
    AS :MODULE_PATHNAME, 'test_privacy' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
\c single :ROLE_DEFAULT_PERM_USER

SHOW timescaledb.telemetry_level;
SET timescaledb.telemetry_level=off;
SHOW timescaledb.telemetry_level;
SELECT _timescaledb_internal.test_privacy();
-- To make sure nothing was sent, we check the UUID table to make sure no exported UUID row was created
SELECT COUNT(*) from _timescaledb_catalog.installation_metadata;

-- Now try to send something
RESET timescaledb.telemetry_level;
-- We don't care about the output of the version checker
SET client_min_messages=error;
SELECT _timescaledb_internal.test_privacy();
RESET client_min_messages;
-- We know it attempted to send if there generated UUIDs
SELECT COUNT(*) from _timescaledb_catalog.installation_metadata where key='uuid';
SELECT COUNT(*) from _timescaledb_catalog.installation_metadata where key='exported_uuid';
SELECT COUNT(*) from _timescaledb_catalog.installation_metadata where key='install_timestamp';
SELECT COUNT(*) from _timescaledb_catalog.installation_metadata;
