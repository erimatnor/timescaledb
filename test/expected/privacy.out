\c single :ROLE_SUPERUSER
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_privacy() RETURNS VOID
    AS :MODULE_PATHNAME, 'test_privacy' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
\c single :ROLE_DEFAULT_PERM_USER
SHOW timescaledb.telemetry_level;
 timescaledb.telemetry_level 
-----------------------------
 basic
(1 row)

SET timescaledb.telemetry_level=off;
SHOW timescaledb.telemetry_level;
 timescaledb.telemetry_level 
-----------------------------
 off
(1 row)

SELECT _timescaledb_internal.test_privacy();
 test_privacy 
--------------
 
(1 row)

-- To make sure nothing was sent, we check the UUID table to make sure no exported UUID row was created
SELECT COUNT(*) from _timescaledb_catalog.installation_metadata;
 count 
-------
     2
(1 row)

-- Now try to send something
RESET timescaledb.telemetry_level;
-- We don't care about the output of the version checker
SET client_min_messages=error;
SELECT _timescaledb_internal.test_privacy();
 test_privacy 
--------------
 
(1 row)

RESET client_min_messages;
-- We know it attempted to send if there generated UUIDs
SELECT COUNT(*) from _timescaledb_catalog.installation_metadata where key='uuid';
 count 
-------
     1
(1 row)

SELECT COUNT(*) from _timescaledb_catalog.installation_metadata where key='exported_uuid';
 count 
-------
     1
(1 row)

SELECT COUNT(*) from _timescaledb_catalog.installation_metadata where key='install_timestamp';
 count 
-------
     1
(1 row)

SELECT COUNT(*) from _timescaledb_catalog.installation_metadata;
 count 
-------
     3
(1 row)

